plugins/org.jboss.tools.openshift.common.ui/src/org/jboss/tools/openshift/internal/common/ui/explorer/BaseExplorerContentProvider.java

```diff
@@ -26,63 +24,52 @@
 import org.eclipse.jface.viewers.TreeViewer;
 import org.eclipse.jface.viewers.Viewer;
 import org.eclipse.swt.widgets.Control;
-import org.jboss.tools.openshift.common.core.connection.ConnectionsRegistry;
 import org.jboss.tools.openshift.common.core.connection.IConnection;
-import org.jboss.tools.openshift.common.core.connection.IConnectionsRegistryListener;
 import org.jboss.tools.openshift.internal.common.ui.OpenShiftCommonUIActivator;
 
 /**
- * Base content provider to hold common logic for OpenShift Explorer contributions
+ * Base content provider to hold common logic for OpenShift Explorer
+ * contributions
  */
-public abstract class BaseExplorerContentProvider implements ITreeContentProvider{
+public abstract class BaseExplorerContentProvider implements ITreeContentProvider {
 
 	private static final String MSG_LOADING_RESOURCES = "Loading OpenShift resources...";
-	private static final Collection<String> PROPERTY_BLACKLIST = Collections.unmodifiableList(Arrays.asList("token"));
 
 	private TreeViewer viewer;
-	private ConnectionsRegistry input;
-	private IConnectionsRegistryListener connectionListener = new ConnectionsRegistryListener();
 
 	// Keep track of what's loading and what's finished
 	private Map<Object, LoadingStub> loadedElements = new ConcurrentHashMap<>();
 	private Map<Object, LoadingStub> loadingElements = new ConcurrentHashMap<>();
-	
+
 	/**
-	 * Get the root elements for the explorer.  This should provide what
-	 * getElements would normally provide but  this method will be called from
-	 * {@link BaseExplorerContentProvider#getElements(Object)} 
+	 * Get the root elements for the explorer. This should provide what
+	 * getElements would normally provide but this method will be called from
+	 * {@link BaseExplorerContentProvider#getElements(Object)}
+	 * 
 	 * @param parentElement
 	 * @return
 	 */
 	protected abstract Object[] getExplorerElements(final Object parentElement);
-	
+
 	/**
-	 * Retrieves the child elements for a given parent.  This should provide what
-	 * getChildren would normally provide but this method will be called from 
+	 * Retrieves the child elements for a given parent. This should provide what
+	 * getChildren would normally provide but this method will be called from
 	 * {@link BaseExplorerContentProvider#getChildren(Object)} or a job for the
 	 * defered loading
+	 * 
 	 * @param parentElement
 	 * @return
 	 */
 	protected abstract Object[] getChildrenFor(Object parentElement);
 
-	
 	@Override
 	public void inputChanged(Viewer viewer, Object oldInput, Object newInput) {
 		this.viewer = (TreeViewer) viewer;
-		if(input != null && connectionListener != null) {
```

Comment:
I see that this gets overriden by the v2 (https://github.com/jbosstools/jbosstools-openshift/pull/1217/files#diff-fec63e7aa9d5165cb82c461b22a3ae5bR75) and v3 (https://github.com/jbosstools/jbosstools-openshift/pull/1217/files#diff-fe25f5ebed5fcd00f5f86c6e04e1aba0R166) content provider subclasses. This portion of the code matching what's now in v2, while v3